include/target.mk

```diff
@@ -38,10 +38,30 @@ DEFAULT_PACKAGES+=procd-ujail
 endif
 
 # mbedTLS wireless features handling
+ifeq ($(CONFIG_TLS_PROVIDER_MBEDTLS),y)
 DEFAULT_PACKAGES+=libustream-mbedtls
 PACKAGE_NO_WIRELESS:=-wpad-basic-mbedtls
-ifneq($(CONFIG_WIRELESS_SUPPORT),)
-DEFAULT_PACKAGES+=wpad-basic-mbedtls
+  ifneq ($(CONFIG_WIRELESS_SUPPORT),)
+    DEFAULT_PACKAGES+=wpad-basic-mbedtls
+  endif
+endif
+
+# OpenSSL and wireless features handling
+ifeq ($(CONFIG_TLS_PROVIDER_OPENSSL),y)
+DEFAULT_PACKAGES+=libustream-openssl
+PACKAGE_NO_WIRELESS:=-wpad-basic-openssl
+  ifneq ($(CONFIG_WIRELESS_SUPPORT),)
+    DEFAULT_PACKAGES+=wpad-basic-openssl
```

Comment:
I think we could use wpad-openssl rather than wpad-basic-openssl on grand flash targets. This would enable more features such as 802.11v/r.

Reply:
Ok, once this gets merged feel free to propose that change via separate pull request/patch.